pipelineutilities/build/lib/pipelineutilities/validate_json.py

```diff
@@ -0,0 +1,248 @@
+# validate_json.py
+""" Common code to retrieve the nd_json_schema and to perform basic json validation against any json schema"""
+
+from jsonschema import validate, ValidationError
+from sentry_sdk import capture_exception
+
+
+def validate_json(json_to_test, schema_to_use, print_error_message=False):
+    """ Validate json against a schema """
+    results = False
+    try:
+        validate(instance=json_to_test, schema=schema_to_use)
+        results = True
+    except ValidationError as e:
+        results = False
+        if print_error_message:
+            print("Validate_json failed with this error message: ", str(e))
+        capture_exception(e)
+    return results
+
+
+nd_json_schema = {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "description": "Schema for validating ND.json",
+    "title": "Tester for ND.json schema",
+    "id": "nd.schema.json",
+    "type": "object",
+    "properties": {
+        "id": {"type": "string"},
+        "sourceSystem": {"type": "string"},
+        "repository": {"type": "string"},
+        "collectionId": {"type": "string"},
+        "parentId": {"type": "string"},
+        "level": {"type": "string"},
+        "title": {"type": "string"},
+        "dateCreated": {
```

Comment:
When I see anyOf, I want to see examples. What would be some numbers? What would be some strings. From the documentation of JSON Schema, an example key/value pair would be useful.

```json
{
  "title" : "Match anything",
  "description" : "This is a schema that matches anything.",
  "default" : "Default value",
  "examples" : [
    "Anything",
    4035
  ]
}
```

Reply:
AnyOf is only there temporarily to accommodate a transition from some of our historical nd.json having strings, and new nd.json having arrays of objects.  Once we're done with the transition, we will remove the strings.

Reply:
Because we're putting this in pipeline utilities, having a separate json file to hold the schema is difficult to accomplish - given path resolution issues in python.  I started with actually multiple separate files, each only containing each object definition, with a master referencing the others.  This turned out to be too problematic, so I merged them all into a single file, and ultimately included the definition in a python module.

Reply:
Can you add this temporary solution as a description. It helps give a reader context.

Reply:
And ugh on the path resolution issues in python. I would really like to have a separate schema file that we can point people to and say: This is what you need to conform to. It lends itself to a more transparent implementation. However, this is not a blocker. It is a very nice to have from a "building a community" perspective.